### PR TITLE
Initial PowerVR GPU support for Vulkan backend

### PR DESCRIPTION
**Pull request by abdelaziz-mahdy**
## Summary

Initial work to add PowerVR GPU support to the Vulkan backend. This PR adds device detection, workgroup tuning, and correctness fixes needed for PowerVR's TBDR architecture.

FP32 inference works correctly with these changes. FP16 has a known issue (see below).

## Motivation

The Vulkan backend currently has no awareness of PowerVR GPUs. Testing on Pixel 10 Pro (PowerVR D-Series DXT-48-1536 MC1) showed that without explicit PowerVR handling, inference produces incorrect results. This PR adds the foundational support needed to run models on PowerVR hardware.

### Device Detection (`vk_api/Device.h`, `vk_api/Device.cpp`)

- Added `POWERVR` to `DeviceType` enum
- Added PowerVR string detection in `PhysicalDevice` constructor

### Compute Graph (`graph/ComputeGraph.h`)

- Added `device_is_powervr()` convenience method

### Convolution Optimization (`graph/ops/impl/Convolution.cpp`)

- Added PowerVR-specific workgroup sizes (32 instead of 64) for convolution dispatch via `conv2d_local_wg_size` callback
- PowerVR GPUs have different execution unit configurations than Adreno/Mali

### Tiling Fix (`api/Context.cpp`)

- Force optimal tiling on PowerVR (linear tiling may produce incorrect results in compute shaders on TBDR architecture)

### Buffer Access (`vk_api/Adapter.cpp`)

- Enable `robustBufferAccess` on PowerVR for well-defined out-of-bounds behavior

## Test Results (Pixel 10 Pro)

I tested 13 minimal single-operator models to isolate behavior:

| Test | Type | Expected | Actual | Status |
|------|------|----------|--------|--------|
| bare_add (x+1.0) | FP16 | 2.0 | 2.0 | PASS |
| bare_half (x*0.5) | FP16 | 0.5 | 0.5 | PASS |
| zero_conv (all-zero weights, no bias) | FP16 | 0.0 | 0.5 | FAIL (+0.5) |
| identity_conv (1x1, input=5.0) | FP16 | 5.0 | 5.5 | FAIL (+0.5) |
| bare_conv2d (3->4, 3x3, no bias) | FP16 | 0.8403 | 1.3400 | FAIL (+0.5) |
| bare_linear (16->4) | FP16 | 1.6 | 2.0996 | FAIL (+0.5) |
| **bare_conv2d_fp32** (3->4, 3x3, no bias) | **FP32** | **0.8403** | **0.8403** | **PASS** |
| bare_conv2d_xnnpack (CPU reference) | FP32 | 0.8403 | 0.8403 | PASS |

**Key observations:**
- **FP32 works correctly** on PowerVR with these changes
- **FP16 conv/linear ops have a deterministic +0.5 offset** (same every run, not random)
- **Non-conv FP16 ops work** (bare_add, bare_half produce correct results)
- **All conv variants affected** — regular, depthwise, pointwise, and linear

MobileNet progressive slices all produce NaN, likely cascading from the +0.5 offset through batch normalization.

### FP16 +0.5 Offset (Unresolved)

All FP16 convolution and linear operations on PowerVR show a constant +0.5 added to the output. This is deterministic and affects all conv variants. The root cause has not been identified yet.

What I've ruled out:
- **Not uninitialized memory**: The offset is deterministic (+0.5 every run), not random
- **Not a convolution algorithm bug**: FP32 convolution produces correct results
- **Not specific to conv size/type**: Affects regular, depthwise, pointwise, and linear

The issue likely lives somewhere in the FP16 prepack or texture path, but I haven't pinpointed it yet. Help from the Vulkan team would be appreciated.

## Test Plan

- [x] Tested 13 minimal single-operator models on Pixel 10 Pro (PowerVR)
- [x] FP32 convolution passes all tests
- [x] Non-conv FP16 ops (add, multiply) pass all tests
- [x] FP16 conv/linear has known +0.5 offset (unresolved)
- [ ] Run existing Vulkan test suite on Adreno/Mali to verify no regressions

## Related Issues

- #17299 - Vulkan backend issues on PowerVR GPUs
- #17341 - StagingBuffer vmaFlushAllocation fixes (separate PR)

cc @SS-JIA @manuelcandales @digantdesai @cbilgin

### CODE REVIEW

File: backends/vulkan/runtime/api/containers/StagingBuffer.h [hunk 1]
```diff
@@ -88,6 +88,11 @@ class StagingBuffer final {
     for (size_t i = 0; i < numel; ++i) {
       dst[i] = static_cast<DST_T>(src[i]);
     }
+    vmaFlushAllocation(
```
Comment (SS-JIA) [inline backends/vulkan/runtime/api/containers/StagingBuffer.h]: Great catch. I noticed that there were a few other calls missing as well, so I created a dedicated PR to fix: https://github.com/pytorch/executorch/pull/17341

Feel free to remove these changes from this PR; also, afaik this doesn't impact the model correctness issue that you are observing.

**Review by SS-JIA — COMMENTED** [review]
Overall, the changes LGTM. Feel free to take it out of draft mode.

File: backends/vulkan/runtime/graph/ops/impl/Convolution.cpp [hunk 2]
```diff
@@ -523,18 +542,21 @@ void add_conv2d_node(
...
       local_wg_size_y = 2;
     }
-    local_wg_size = {64 / local_wg_size_y, local_wg_size_y, 1};
+    local_wg_size = {max_local_size / local_wg_size_y, local_wg_size_y, 1};
   } else if (method == Conv2dMethod::Depthwise) {
-    local_wg_size = {64, 1, 1};
+    local_wg_size = {max_local_size, 1, 1};
   } else {
     local_wg_size = graph.create_local_wg_size(wg_size);
   }
```
[the remark this answers is not in the document]
Comment (SS-JIA) [reply backends/vulkan/runtime/graph/ops/impl/Convolution.cpp]: Fair point - I know this was an issue before you made these changes, but would you mind applying these suggested changes in your PR? Much appreciated 🙏

**Review by SS-JIA — CHANGES_REQUESTED** [review]
Overall LGTM, just a very small request.

File: backends/vulkan/runtime/graph/ops/impl/Convolution.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (abdelaziz-mahdy) [reply backends/vulkan/runtime/graph/ops/impl/Convolution.cpp]: done thanks.
